packages/next-swc/crates/core/tests/fixture/optimize_server_react/1/input.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useLayoutEffect, useMemo } from 'react'
+import React from 'react'
+
+export default function App() {
+  useEffect(() => {
+    console.log('Hello World')
+  }, [])
+
+  useLayoutEffect(() => {
+    function foo() {}
+    return () => {}
+  }, [1, 2, App])
```

Comment:
Does this care as to whether the dependency array has side effects? As in
```
useEffect(() => {}, [runSideEffect()])

Reply:
Very good call, I think that should be banned but I'll make sure to opt-out in that case.

Reply:
Done, it now will opt-out these cases:

```js
  useLayoutEffect(() => {}, [runSideEffect()])
  useEffect(() => {}, [1, runSideEffect(), 2])
  useEffect(() => {}, getArray())
```